### PR TITLE
refactor(all): move stakePoolTransformer, CoinId, formatPercentages, …

### PR DESCRIPTION
**Pull request by szymonmaslowski**
…getRandomIcon outside extension

# Checklist

- [x] JIRA - [\<link>](https://input-output.atlassian.net/browse/LW-6678)
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

The main intention was to move the `stakePoolTransformer` util to the shared package in order to make it possible to be reused outside the extension code.
In order to achieve that it was required to move also other entities which were the `stakePoolTransformer`'s dependencies.

Resulting destinations:
 - `stakePoolTransformer`: `@lace/cardano` (`Wallet.util.stakePoolTransformer`)
 - `CoinId`: `@lace/cardano` (`Wallet.CoinId`)
 - `formatPercentages`: `@lace/common`
 - `getRandomIcon`: `@lace/common`

## Testing

Everything should work as previously, app should be functional, tests are green

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@d1terqjryk7mu9.cloudfront.net/linux/chrome/608/5355965832/index.html) for [cc0a84db](https://github.com/input-output-hk/lace/pull/167/commits/cc0a84dbf74259e579c40ee4b25e26e88f74949d)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 33     | 3      | 0       | 0     | 36    | ❌     |

### CODE REVIEW

**Review by mrcnk — APPROVED** [review]

**Review by przemyslaw-wlodek — APPROVED** [review]

File: apps/browser-extension-wallet/src/providers/currency/store.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import create, { UseStore } from 'zustand';
 import { Wallet } from '@lace/cardano';
 import { currencyCode, currencyMap, defaultCurrency, currencies } from './constants';
-import { CoinId, CurrencyInfo } from '../../types';
+import { CurrencyInfo } from '../../types';
```
Comment (mchappell) [inline apps/browser-extension-wallet/src/providers/currency/store.ts:4]:
```suggestion
import { CurrencyInfo } from '@src/types';
```

File: apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/components/SendTransactionSummary.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import React, { useMemo } from 'react';
 import isUndefined from 'lodash/isUndefined';
 import { useWalletStore } from '../../../../../stores';
-import { CardanoTxOut, CoinId, CurrencyInfo, TokensDetails } from '../../../../../types';
+import { CardanoTxOut, CurrencyInfo, TokensDetails } from '../../../../../types';
```
Comment (mchappell) [inline apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/components/SendTransactionSummary.tsx:4]:
```suggestion
import { CardanoTxOut, CurrencyInfo, TokensDetails } from '@src/types';
```

File: packages/common/package.json [hunk 3]
```diff
@@ -38,8 +38,10 @@
     "watch": "yarn build --watch"
   },
   "dependencies": {
+    "@cardano-sdk/util": "~0.11.0",
     "antd": "^4.17.3",
     "classnames": "^2.3.1",
+    "jdenticon": "3.1.0",
```
Comment (mchappell) [inline packages/common/package.json:44]: If it's been added here, can it be removed from `apps/browser-extension-wallet` and reference all `getRandomIcon` from a single file.
To do this you'll need to remove the method from `packages/staking/src/features/overview/staking-info-card/StakingInfoCard.tsx`

**Review by mchappell — APPROVED** [review]

**Comment by przemyslaw-wlodek** [comment]
Fix for smoke tests will arrive with [this PR](https://github.com/input-output-hk/lace/pull/175). Merging 🎉

**Pull request merged**